lib/msal-browser/src/interaction_client/StandardInteractionClient.ts

```diff
@@ -39,6 +39,25 @@ export abstract class StandardInteractionClient extends BaseInteractionClient {
         this.logger.verbose("initializeAuthorizationRequest called", request.correlationId);
         const generatedPkceParams = await this.browserCrypto.generatePkceCodes();
 
+        // Generate Session Transport Key if Refresh Token Binding is enabled
+        if (this.config.system.refreshTokenBinding) {
+            this.logger.verbose("Refresh token binding enabled, attempting to generate Session Transport Key");
+            try {
+                const sessionTransportKeyThumbprint = await this.browserCrypto.getPublicKeyThumbprint(request, CryptoKeyTypes.StkJwk);
+                request.stkJwk = sessionTransportKeyThumbprint;
+                this.logger.verbose("Successfully generated and stored Session Transport Key");
+            } catch(error) {
+                if(error instanceof BrowserAuthError) {
+                    if (error.errorCode === "key_generation_failed") {
+                        this.logger.error("Failed to generate Session Transport Key, refresh token will be unbound.");
```

Comment:
Are we swallowing this error? If so, should we just log these messages where the errors are thrown instead of throwing the error and catching?

Reply:
@tnorling The reason I did it this way is that these messages are bound RT specific but the error throwing method here (getPublicKeyThumbprint) is also used for AT PoP key generation (so messages have to stay generic). We do "swallow" the error because we want to smoothly fall back to unbound RT if key generation fails instead of aborting the auth request.

What do you think?

Reply:
Gotcha, thanks for the context. Can you add a brief comment explaining why we're ignoring these particular errors and then we also need to rethrow any error that doesn't match your conditionals here

Reply:
Addes comments and refactored the logic to make it cleaner here